test/solver_tests/neighborjoining_solver_test.py

```diff
@@ -124,13 +124,57 @@ def delta_fn(
 
 
     def test_constructor(self):
-
         self.assertIsNone(self.nj_solver.dissimilarity_function)
         self.assertIsNotNone(self.nj_solver_delta.dissimilarity_function)
         self.assertIsNotNone(self.basic_tree.get_dissimilarity_map())
 
-    def test_compute_q(self):
+        nothing_solver = cas.solver.NeighborJoiningSolver(
+            dissimilarity_function=None, add_root=False
+        )
+
+        no_root_tree = cas.data.CassiopeiaTree(
+            character_matrix=self.cm, dissimilarity_map=self.basic_dissimilarity_map, 
+        )
+
+        with self.assertRaises(Exception) as context:
+            nothing_solver.solve(no_root_tree)
 
+        self.assertTrue(
+            "Please specify an explicit root sample in the Cassiopeia Tree or specify the solver to add an implicit root"
+            in str(context.exception)
+        )
```

Comment:
This is a dangerous test -- what if we change the error message? You can just leave it as `self.assertRaises(DistanceSolverException, nothing_solver.solve, no_root_tree)`